datafusion/expr/src/utils.rs

```diff
@@ -579,6 +580,13 @@ pub fn from_plan(
 
             Ok(plan.clone())
         }
+        LogicalPlan::Prepare(Prepare {
+            name, data_types, ..
+        }) => Ok(LogicalPlan::Prepare(Prepare {
+            name: name.clone(),
+            data_types: data_types.clone(),
+            input: Arc::new(inputs[0].clone()),
+        })),
```

Comment:
Is it possible and allowed here that the method passed in a totally different input plan ?

Reply:
With the code we implement, the answer is no unless there are bugs